django/db/models/base.py

```diff
@@ -2063,6 +2063,24 @@ def _check_constraints(cls, databases):
                         id='models.W044',
                     )
                 )
+            if not (
+                connection.features.supports_tablespaces or
+                'supports_tablespaces' in cls._meta.required_db_features
+            ) and any(
+                isinstance(constraint, UniqueConstraint) and constraint.db_tablespace
+                for constraint in cls._meta.constraints
+            ):
+                errors.append(
+                    checks.Warning(
+                        '%s does not support setting a custom tablespace.' % connection.display_name,
+                        hint=(
+                            "A constraint won't be created. Silence this "
+                            "warning if you don't care about it."
+                        ),
+                        obj=cls,
+                        id='models.W045',
+                    )
+                )
```

Comment:
IMO this check is not necessary, we should simply ignore `db_tablespace` when a database doesn't support it.